lxd/device/disk.go

```diff
@@ -172,6 +172,7 @@ func (d *disk) validateConfig(instConf instance.ConfigReader) error {
 		"ceph.user_name":    validate.IsAny,
 		"boot.priority":     validate.Optional(validate.IsUint32),
 		"path":              validate.IsAny,
+		"io.cache":          validate.Optional(validate.IsOneOf("none", "writeback", "unsafe")),
```

Comment:
As this is a VM specific config setting, we should add validation for that, similar to:

https://github.com/lxc/lxd/blob/master/lxd/device/proxy.go#L100-L102

Or dynamically set the rules like:

https://github.com/lxc/lxd/blob/master/lxd/device/tpm.go#L39-L45

Keeping in mind that we should support all options when the instance type is `instancetype.Any` for profile validation.